### PR TITLE
Adds apicula toolchain to gowin platform

### PR DESCRIPTION
**Pull request by Mai-Lapyst**
This PR:
- Adds GowinApiculaToolchain by using yosys + nextpnr
- Moves constraints generation from GowinToolchain.build_io_constraints to _build_cst to be reuseable
- Modifies NextPNRWrapper.get_call so command only contains flag for constraint fmt if it is non-empty. This is necessary as the `nextpnr-himbaechel` needs to be used, where the constaints file needs to be specified via `--vopt cst=file`, so the intended way of NextPNRWrapper for constraint-passing does not work anymore.

### CODE REVIEW

File: litex/build/gowin/apicula.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    pnr_fmt    = "report"
+    packer_cmd = "gowin_pack"
+
+    def __init__(self):
+        super().__init__()
+        self.additional_cst_commands = []
+
+    def build_timing_constraints(self, vns):
+        # nextpnr-himbaechel for gowin currently doesnt supports timing constaints
+        pass
```
Comment (trabucayre) [inline litex/build/gowin/apicula.py:18]: Not required since implemented by `GenericToolchain` module (and must return `("","")` when unused).

**Comment by trabucayre** [comment]
LGTM.
Just a small remark: `build_timing_constraints` must be removed to let default behavior or a return must be added.
Thanks!

**Comment by Mai-Lapyst** [comment]
> LGTM. Just a small remark: `build_timing_constraints` must be removed to let default behavior or a return must be added. Thanks!

Thanks for the response! Have removed the method you mentioned and also added the missing license header to apicula.py

**Comment by trabucayre** [comment]
Applied.
Thanks @Mai-Lapyst

**Pull request merged**